internal/idp/providers/ldap/session.go

```diff
@@ -47,13 +71,10 @@ func (s *Session) FetchUser(_ context.Context) (idp.User, error) {
 			s.Password,
 			s.Provider.timeout,
 		)
-		if err == nil && user != nil {
-			break
+		if err != nil {
+			return nil, err
 		}
 	}
-	if err != nil {
-		return nil, err
-	}
 
 	var emailVerified bool
```

Comment:
i would put the mapping into a separate method, so you're able to test it as well